src/compiler/compile/nodes/Style.ts

```diff
@@ -0,0 +1,24 @@
+import Node from './shared/Node';
+import Expression from './shared/Expression';
+import { TemplateNode } from '../../interfaces';
+import TemplateScope from './shared/TemplateScope';
+import Component from '../Component';
+
+export default class Style extends Node {
+	type: 'Style';
+	name: string;
+	expression: Expression;
+	text: string;
+
+	constructor(component: Component, parent: Node, scope: TemplateScope, info: TemplateNode) {
+		super(component, parent, scope, info);
+
+		this.name = info.name;
+
+		this.text = info.text || null;
```

Comment:
Style directives need to take strings as values (unlike other directives, which always take expressions), so I tacked on this `text` property to the style directive node.